src/cocotb_bus/compat.py

```diff
@@ -15,17 +16,112 @@ def TestFactory(*args, **kwargs):
         return cocotb.regression.TestFactory(*args, **kwargs)
 
 
-if parse_version(cocotb.__version__) > parse_version("2.dev0"):
+cocotb_2x_or_newer = parse_version(cocotb.__version__) > parse_version("2.dev0")
+if cocotb_2x_or_newer:
+    from cocotb.types import LogicArray, Range
+    BinaryType = LogicArray
+
+
     def set_event(event, data):
         event.data = data
         event.set()
 
     def coroutine(f):
         return f
 
+
+    def create_binary(binstr: Union[int, str, bytes, LogicArray], bit_count: int, big_endian: bool):
+        if big_endian:
+            range=Range(0, 'to', bit_count - 1)
+        else:
+            range=Range(bit_count - 1, 'downto', 0)
+
+        if isinstance(binstr, bytes):
+            if not big_endian:
+                binstr = reversed(binstr)
+            binstr = "".join([format(char, "08b") for char in binstr])
+
+            if big_endian:
+                binstr = binstr + "0" * (bit_count - len(binstr))
+            else:
+                binstr = "0" * (bit_count - len(binstr)) + binstr
+
+        return LogicArray(binstr, range=range)
+
+
+    def create_binary_from_other(value: BinaryType, binstr: Union[int, str, bytes]):
+        return LogicArray(value=binstr, range=value.range)
+
+
+    def convert_binary_to_bytes(value: BinaryType, big_endian: bool):
+        return value.to_bytes('big' if big_endian else 'little')
+
+
+    def convert_binary_to_unsigned(value: Union[int, BinaryType]):
+        if isinstance(value, int):
+            return value
+        return value.to_unsigned()
+
+
+    def convert_binary_to_binstr(value: BinaryType):
+        return str(value)
```

Comment:
This doesn't seem useful. `str()` casts work on both, right?

Reply:
Indeed, fixed now. Thanks.